cmd/umoci/insert.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * umoci: Umoci Modifies Open Containers' Images
+ * Copyright (C) 2018 Cisco
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/apex/log"
+	"github.com/openSUSE/umoci/mutate"
+	"github.com/openSUSE/umoci/oci/cas/dir"
+	"github.com/openSUSE/umoci/oci/casext"
+	igen "github.com/openSUSE/umoci/oci/config/generate"
+	"github.com/openSUSE/umoci/oci/layer"
+	ispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+var insertCommand = uxHistory(cli.Command{
+	Name:  "insert",
+	Usage: "insert a file into an OCI image without unpacking/repacking it",
+	ArgsUsage: `--image <image-path>[:<tag>] <file> <path>
+
+Where "<image-path>" is the path to the OCI image, "<tag>" is the name of the
+tag that the content wil be inserted into (if not specified, defaults to
+"latest"), "<file>" is the file or folder to insert, and "<path>" is the prefix
+inside the image to insert it into.`,
+
+	Category: "image",
+
+	Action: insert,
+
+	Before: func(ctx *cli.Context) error {
+		if ctx.NArg() != 2 {
+			return errors.Errorf("invalid number of positional arguments: expected <file> and <path>")
+		}
+		if ctx.Args()[0] == "" {
+			return errors.Errorf("path cannot be empty")
+		}
+		if ctx.Args()[1] == "" {
+			return errors.Errorf("path cannot be empty")
+		}
```

Comment:
I would specify which path cannot be empty. Also, I'm not sure why the prefix cannot be empty -- an empty prefix just means it should be added to the root, no?